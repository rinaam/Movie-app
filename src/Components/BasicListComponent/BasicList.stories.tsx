import React from "react";
import { Story, Meta } from "@storybook/react";

import BasicListComponent, { IBasicListProps } from "./BasicListComponent";

export default {
  title: "BasicList",
  component: BasicListComponent,
} as Meta;

const Template: Story<IBasicListProps> = (args) => (
  <BasicListComponent {...args} />
);

export const Basic = Template.bind({});
Basic.args = {
  list: [
    {
      title: "title1",
      sufix: "item1",
    },
    {
      title: "title2",
      sufix: "item2",
    },
    {
      title: "title3",
      sufix: "item3",
    },
  ],
};
