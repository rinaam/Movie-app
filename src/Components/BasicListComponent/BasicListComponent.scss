@use "sass:map";
@import "../../Styles/variables.scss";
@import "../../Styles/breakpoints.scss";

.basic-list {
  line-height: 1.5;
  @include breakpoint(smallDevices) {
    margin: map.get($spacing, "l") 0px;
  }
}
