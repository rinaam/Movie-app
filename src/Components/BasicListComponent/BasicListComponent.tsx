import React from "react";
import "./BasicListComponent.scss";
import BasicListItemComponent from "../BasicListItemComponent/BasicListItemComponent";

export interface IBasicListProps {
  list: TBasicListItem[];
}

export type TBasicListItem = {
  title: string;
  sufix: string;
};

const BasicListComponent: React.FC<IBasicListProps> = ({ list }) => (
  <div className="basic-list">
    {list.map((item) => (
      <BasicListItemComponent
        title={item.title}
        sufix={item.sufix}
        key={item.title}
      />
    ))}
  </div>
);

export default BasicListComponent;
