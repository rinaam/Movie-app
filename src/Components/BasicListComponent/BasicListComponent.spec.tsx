import { create } from "react-test-renderer";
import BasicListComponent from "./BasicListComponent";
import React from "react";

describe("BasicListComponent", () => {
  const fakeList = [
    {
      title: "title",
      sufix: "item",
    },
  ];
  it("should render BasicListComponent", () => {
    const wrapper = create(<BasicListComponent list={fakeList} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
