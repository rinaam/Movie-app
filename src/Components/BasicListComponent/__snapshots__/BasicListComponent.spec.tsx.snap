// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicListComponent should render BasicListComponent 1`] = `
<div
  className="basic-list"
>
  <div
    className="basic-list-item"
  >
    <p>
      title
    </p>
    <p>
      item
    </p>
  </div>
</div>
`;
